chacra/models/binaries.py

```diff
@@ -52,6 +52,12 @@ def __init__(self, name, project, repo=None, **kw):
             if key in kw.keys():
                 setattr(self, key, kw[key])
         self.repo = repo or self._get_or_create_repo()
+        # ensure that the repo.type is set
+        self._set_repo_type(self.repo)
```

Comment:
Why pass self.repo here? Couldn't you use ``self.repo`` below in ``__set_repo_type`` and avoid passing the arg?